src/game_control.rs

```diff
@@ -0,0 +1,35 @@
+use bevy::prelude::{Input, KeyCode, Res};
+
+pub enum GameControl {
+    Up,
+    Down,
+    Left,
+    Right,
+}
+
+impl GameControl {
+    pub fn pressed(&self, keyboard_input: &Res<Input<KeyCode>>) -> bool {
+        match self {
+            GameControl::Up => {
+                keyboard_input.pressed(KeyCode::W) || keyboard_input.pressed(KeyCode::Up)
+            }
+            GameControl::Down => {
+                keyboard_input.pressed(KeyCode::S) || keyboard_input.pressed(KeyCode::Down)
+            }
+            GameControl::Left => {
+                keyboard_input.pressed(KeyCode::A) || keyboard_input.pressed(KeyCode::Left)
+            }
+            GameControl::Right => {
+                keyboard_input.pressed(KeyCode::D) || keyboard_input.pressed(KeyCode::Right)
+            }
+        }
+    }
+}
+
+pub fn is_pressed(control: GameControl, input: &Res<Input<KeyCode>>) -> f32 {
```

Comment:
Perhaps this function could have a more descriptive name,`is_pressed` can be misleading and people might think it returns a boolean, though I myself don't mind since my editor makes the return typing very visible. What do you think?

Reply:
Maybe `movement`?

Reply:
Thanks! What do you think about `get_movement`?